cpp/tests/experimental/katz_centrality_test.cpp

```diff
@@ -311,35 +260,48 @@ class Tests_KatzCentrality : public ::testing::TestWithParam<KatzCentrality_Usec
   }
 };
 
+using cugraph::test::File_Usecase;
+using cugraph::test::Rmat_Usecase;
+
+using Tests_KatzCentrality_File = Tests_KatzCentrality<File_Usecase>;
+using Tests_KatzCentrality_Rmat = Tests_KatzCentrality<Rmat_Usecase>;
+
 // FIXME: add tests for type combinations
-TEST_P(Tests_KatzCentrality, CheckInt32Int32FloatFloat)
+TEST_P(Tests_KatzCentrality_File, CheckInt32Int32FloatFloat)
+{
+  auto param = GetParam();
+  run_current_test<int32_t, int32_t, float, float>(std::get<0>(param), std::get<1>(param));
+}
+
+TEST_P(Tests_KatzCentrality_Rmat, CheckInt32Int32FloatFloat)
 {
-  run_current_test<int32_t, int32_t, float, float>(GetParam());
+  auto param = GetParam();
+  run_current_test<int32_t, int32_t, float, float>(std::get<0>(param), std::get<1>(param));
 }
 
 INSTANTIATE_TEST_CASE_P(
   simple_test,
-  Tests_KatzCentrality,
+  Tests_KatzCentrality_File,
+  ::testing::Values(
```

Comment:
For this and all others, can we use [`::testing::Combine()`](https://code.ornl.gov/ezx/random_ezx/-/blob/3f7be7147cf536b8c7fd9483eaaa8b6f4fbb1671/gtest/include/gtest/gtest-param-test.h#L1225) to auto-generate the input combinations?  I believe it may require you to handle a `std::tuple` instead of a pair though.

Reply:
We could.  The question is do we want to test all combinations in the Cartesian product?  Some of the other examples (e.g. BFS and SSSP) require parameters that might not make sense.  For example, testing a graph with 100 vertices but specifying a starting vertex id of 1000 would fail.  But it's reasonable to test that using a starting vertex other than 0 does pass.

I'm inclined to explicitly enumerate them... although I could convert from pair to tuple (already using the `std::get` syntax) and construct an example to validate that the `::testing::Combine()` pattern would work.

Reply:
That's a good point about not always wanting every combo.  Is this something that can be used for tests where all combos are needed, but not for others?  (I was specifically noticing that the inputs for Katz looked like every combination).  I don't have a strong preference but was just thinking it could clean up some tests, your call.

I suppose if you at least refactored it to a pair as you suggested, then other tests could possibly use `Combine()` if it makes sense for them (and possibly be closer to other patterns gtest uses).

Reply:
I'll push an update that uses this where appropriate.